src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -120,7 +126,16 @@
             throw new IOException("Unsupported credential type: " + credentials.getClass().getName());
         }
 
-        return gb.build();
+        GitHub github = gb.build();
+        try {
+            github.checkApiUrlValidity();
```

Comment:
I believe it worth. This can be very helpful (console output):

```
Started by user anonymous
FATAL: It seems https://api.github.com is unreachable

Finished: FAILURE
```